getsolus/packages
# Samba 4.19.6 stack upgrade

**Summary**

- Update from 4.17.x series to 4.19.6 (including talloc, tdb, tevent and ldb libs)
- Performance regressions due to security fixes have been mitigated
- Add 'server addresses' smb.conf parameter
- Migrate smbget to use common command line parser for robustness
- Remove all in-tree cryptography in favour of relying fully on GnuTLS

**Changelog**

Samba 4.18.x:
- [samba-4.18.0](https://www.samba.org/samba/history/samba-4.18.0.html)
- [samba-4.18.1](https://www.samba.org/samba/history/samba-4.18.1.html)
- [samba-4.18.2](https://www.samba.org/samba/history/samba-4.18.2.html)
- [samba-4.18.3](https://www.samba.org/samba/history/samba-4.18.3.html)
- [samba-4.18.4](https://www.samba.org/samba/history/samba-4.18.4.html)
- [samba-4.18.5](https://www.samba.org/samba/history/samba-4.18.5.html)
- [samba-4.18.6](https://www.samba.org/samba/history/samba-4.18.6.html)
- [samba-4.18.7](https://www.samba.org/samba/history/samba-4.18.7.html)
- [samba-4.18.8](https://www.samba.org/samba/history/samba-4.18.8.html)
- [samba-4.18.9](https://www.samba.org/samba/history/samba-4.18.9.html)
- [samba-4.18.10](https://www.samba.org/samba/history/samba-4.18.10.html)
- [samba-4.18.11](https://www.samba.org/samba/history/samba-4.18.11.html)

Samba 4.19.x:
- [samba-4.19.0](https://www.samba.org/samba/history/samba-4.19.0.html)
- [samba-4.19.1](https://www.samba.org/samba/history/samba-4.19.1.html)
- [samba-4.19.2](https://www.samba.org/samba/history/samba-4.19.2.html)
- [samba-4.19.3](https://www.samba.org/samba/history/samba-4.19.3.html)
- [samba-4.19.4](https://www.samba.org/samba/history/samba-4.19.4.html)
- [samba-4.19.5](https://www.samba.org/samba/history/samba-4.19.5.html)
- [samba-4.19.6](https://www.samba.org/samba/history/samba-4.19.6.html)

**Test Plan**

- Run `sudo testparm -v` and check for anomalies in the default config
- Run `sudo systemctl enable --now smb ; smbclient -N -L 127.0.0.1`, and
check that the service runs, and that it is possible to connect to the
smb daemon
- Look for `smbd` and ports 139 and 445 in:
`sudo ss -plantu |grep -E '139|445|smbd'`
- Reboot and ensure that anonymous and authenticated user access works in
the Dolphin (kio) and Nautilus (gvfs) file managers
- Ensure that smb:// URI playback works in:
- `celluloid` (used by Budgie and GNOME)
- `haruna` (used by KDE)
- `kodi`
- `mpv` (unauthenticated access works, authenticated access won't)
- `parole` (used by XFCE)
- `vlc` (note that vlc can be temperamental re. authenticated access)

**Checklist**


Closes getsolus/packages/issues/1469
